examples/templates/posix_sockets.h

```diff
@@ -3,8 +3,10 @@
 
 #include <stdio.h>
 #include <sys/types.h>
+#if !defined(WIN32)
 #include <sys/socket.h>
 #include <netdb.h>
+#endif
```

Comment:
This and the non-blocking special case below are the only win32 specific changes here.
Is this acceptable?
Otherwise the below function will need to be added into the unittest directly

Reply:
I'd prefer this either go in the unittest directly, or as a seperate template.

Reply:
Yeah, this looks ok to me